internal/exec/stages/disks/partitions.go

```diff
@@ -93,39 +93,47 @@ func partitionMatches(existing util.PartitionInfo, spec types.Partition) error {
 }
 
 // partitionShouldBeInspected returns if the partition has zeroes that need to be resolved to sectors.
-func partitionShouldBeInspected(part types.Partition) bool {
+func partitionShouldBeInspected(part sgdisk.Partition) bool {
 	if part.Number == 0 {
 		return false
 	}
-	return (part.StartMiB != nil && *part.StartMiB == 0) ||
-		(part.SizeMiB != nil && *part.SizeMiB == 0)
+	return (part.StartSector != nil && *part.StartSector == 0) ||
+		(part.SizeInSectors != nil && *part.SizeInSectors == 0)
 }
 
-func convertMiBToSectors(mib *int, sectorSize int) {
+func convertMiBToSectors(mib *int, sectorSize int) *int {
 	if mib != nil {
-		*mib = (*mib) * (1024 * 1024 / sectorSize)
+		v := (*mib) * (1024 * 1024 / sectorSize)
+		return &v
+	} else {
+		return nil
 	}
 }
 
 // getRealStartAndSize returns a map of partition numbers to a struct that contains what their real start
 // and end sector should be. It runs sgdisk --pretend to determine what the partitions would look like if
 // everything specified were to be (re)created.
-// It also converts everything to sectors so the StartMiB/SizeMiB will NOT be in MiB after this call
-func (s stage) getRealStartAndSize(dev types.Disk, devAlias string, diskInfo util.DiskInfo) ([]types.Partition, error) {
-	op := sgdisk.Begin(s.Logger, devAlias)
-	for _, part := range dev.Partitions {
-		convertMiBToSectors(part.SizeMiB, diskInfo.LogicalSectorSize)
-		convertMiBToSectors(part.StartMiB, diskInfo.LogicalSectorSize)
+func (s stage) getRealStartAndSize(dev types.Disk, devAlias string, diskInfo util.DiskInfo) ([]sgdisk.Partition, error) {
```

Comment:
Note for review: this function is evil.  The `part` values in each loop are private copies, so field updates don't survive to the next loop.  **Except** for `convertMiBToSectors` in the base commit, which assigns _through_ int pointers rather than replacing them.